lib/axiom/fs/base/file_system.js

```diff
@@ -205,11 +205,16 @@ FileSystem.prototype.readFile = function(
   var whence = opt_whence || SeekWhence.Begin;
 
   return this.createOpenContext(path, openMode).then(
-      function(cx) {
-        return cx.open().then(function() {
-          return cx.read(offset, whence, dataType);
-        });
-  });
+    function(cx) {
+      return OpenContext.scope(cx, function() {
+        return cx.open().then(
+          function() {
```

Comment:
style: This should either join the above line (and unindent the following) or be +2 more spaces as a broken statement.